build/docker/docker-compose.yml

```diff
@@ -24,8 +26,11 @@ services:
       interval: 5s
       timeout: 4s
       retries: 3
+    # "Larger settings for shared_buffers usually require a corresponding increase in max_wal_size, 
+    # to spread out the process of writing large quantities of new or changed data over a longer period of time."
     command: "
-      -c shared_buffers=2GB
+      -c shared_buffers=1GB
```

Comment:
What value do you currently have for wal_buffers? We've found that bumping this up to 16MB improves performance (in our ingestion workload).

Reply:
i'm not setting that currently, but could easily add it.  however, based on https://postgresqlco.nf/doc/en/param/wal_buffers/
> The default setting of -1 selects a size equal to 1/32nd (about 3%) of [shared_buffers](https://postgresqlco.nf/doc/en/param/shared_buffers/)

0.03 * 1 GB = 60 MB, so a setting of 16 MB would actually be reducing it.
The default setting of -1 selects a size equal to 1/32nd (about 3%) of [shared_buffers](https://postgresqlco.nf/doc/en/param/shared_buffers/)

Reply:
From the docs: "The default setting of -1 selects a size equal to 1/32nd (about 3%) of [shared_buffers](https://www.postgresql.org/docs/current/runtime-config-resource.html#GUC-SHARED-BUFFERS), but not less than 64kB nor more than the size of one WAL segment, typically 16MB."

So for larger sizes of shared_buffers, we'll get the 16MB. We were seeing smaller values on some of the database instances we had provisioned, and so setting it manually helped.

Net: in real world scenarios shared_buffers will be sufficiently large enough (>512MB) to ensure wal_buffers hits 16MB.